src/main/java/com/iexec/common/chain/IexecHubAbstractService.java

```diff
@@ -111,11 +110,43 @@ public IexecHubAbstractService(Credentials credentials,
                                    int blockTime,
                                    int nbBlocksToWaitPerRetry,
                                    int maxRetries) {
+        this(
+                credentials,
+                web3jAbstractService,
+                iexecHubAddress,
+                Duration.ofMillis(blockTime > 0 ? blockTime : DEFAULT_BLOCK_TIME),
```

Comment:
I don't think we need that. If block time is not > 0, below base constructor will build duration with DEFAULT_BLOCK_TIME.
I would also add a deprecation tag on this method (comment needs to be updated too, it is not the base constructor anymore).

Reply:
Right, thanks.
I was also thinking about deprecating this constructor but was unsure about that. So thanks again for pointing this out :)